service-registry/src/main/java/io/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -143,12 +146,53 @@ private boolean registerSchemas() {
       boolean exists = srClient.isSchemaExist(microservice.getServiceId(), schemaId);
       LOGGER.info("schemaId {} exists {}", schemaId, exists);
       if (!exists) {
-        if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
-          return false;
+        if (serviceCenterEnvIsDev()) {
+            if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
+                return false;
+            }
+        }
+      } else {
+        //query if schema exist, if not exist, register schema&summary in SC
+        String serviceId = microservice.getServiceId();
+        String schema = srClient.getSchema(serviceId, schemaId);
+        if (null == schema) {
```

Comment:
we already make sure schema is exist, but here we can not get it
can we override remote schema directly?

Reply:
done，there is no need to query schema, if schemaId exist, just query schemaSummary